### PR TITLE
[DNM] add exceeding check for upper_bound

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
For debugging.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -1097,6 +1131,25 @@ name = "rand_core"
 version = "0.2.1"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 
+[[package]]
+name = "rayon"
```
Comment (breeswish) [inline Cargo.lock]: why is rayon introduced?

File: src/storage/mvcc/reader/backward_scanner.rs [hunk 2]
```diff
@@ -180,10 +180,33 @@ impl<S: Snapshot> BackwardScanner<S> {
                     self.upper_bound.as_ref().unwrap(),
                     &mut self.statistics.write,
                 )?;
+
+                if self.write_cursor.valid() {
+                    let key = self.write_cursor.key(&mut self.statistics.write);
+                    if key >= self.upper_bound.as_ref().unwrap().as_encoded().as_slice() {
```
Comment (breeswish) [inline src/storage/mvcc/reader/backward_scanner.rs]: what if upper_bound is None?

**Comment by breeswish** [comment]
unit test failed

**Comment by overvenus** [comment]
It fails as expected because it panics on key not in region.

**Pull request closed without merging**